src/iOS/Xamarin.iOS/Samples/Network Analysis/FindRoute/FindRoute.cs

```diff
@@ -62,154 +62,162 @@ public override void ViewDidLoad()
 
-            // Create the graphics overlay and add the stop graphics
+            // Create the graphics overlay and add the stop graphics.
             _routeGraphicsOverlay = new GraphicsOverlay();
             _routeGraphicsOverlay.Graphics.Add(fromGraphic);
             _routeGraphicsOverlay.Graphics.Add(toGraphic);
 
-            // Get an Envelope that covers the area of the stops (and a little more)
+            // Get an Envelope that covers the area of the stops (and a little more).
             Envelope routeStopsExtent = new Envelope(fromPoint, toPoint);
             EnvelopeBuilder envBuilder = new EnvelopeBuilder(routeStopsExtent);
             envBuilder.Expand(1.5);
 
-            // Create a new viewpoint apply it to the map view when the spatial reference changes
+            // Create a new viewpoint apply it to the map view when the spatial reference changes.
             Viewpoint sanDiegoViewpoint = new Viewpoint(envBuilder.ToGeometry());
             _myMapView.SpatialReferenceChanged += (s, e) => _myMapView.SetViewpoint(sanDiegoViewpoint);
 
-            // Add a new Map and the graphics overlay to the map view
+            // Add a new Map and the graphics overlay to the map view.
             _myMapView.Map = new Map(Basemap.CreateStreetsVector());
             _myMapView.GraphicsOverlays.Add(_routeGraphicsOverlay);
         }
 
-        private async void SolveRouteButtonClick(object sender, EventArgs e)
+        private async void SolveRouteButton_Click(object sender, EventArgs e)
         {
-            // Create a new route task using the San Diego route service URI
+            // Create a new route task using the San Diego route service URI.
             RouteTask solveRouteTask = await RouteTask.CreateAsync(_sanDiegoRouteServiceUri);
 
-            // Get the default parameters from the route task (defined with the service)
+            // Get the default parameters from the route task (defined with the service).
             RouteParameters routeParams = await solveRouteTask.CreateDefaultParametersAsync();
 
-            // Make some changes to the default parameters
+            // Make some changes to the default parameters.
             routeParams.ReturnStops = true;
             routeParams.ReturnDirections = true;
 
-            // Set the list of route stops that were defined at startup
+            // Set the list of route stops that were defined at startup.
             routeParams.SetStops(_routeStops);
 
-            // Solve for the best route between the stops and store the result
+            // Solve for the best route between the stops and store the result.
             RouteResult solveRouteResult = await solveRouteTask.SolveRouteAsync(routeParams);
 
-            // Get the first (should be only) route from the result
+            // Get the first (should be only) route from the result.
             Route firstRoute = solveRouteResult.Routes.FirstOrDefault();
```

Comment:
Best practice opinion: Consider using `Single()` or `First()` instead of `FirstOrDefault()` here. That way the code clearly expresses its assumption ("exactly one" or "at least one" route in results). If the server ever returns an unexpected result, you'd get a specific and helpful error message instead of a generic NullReferenceException somewhere later in the method.